ash/src/extensions/ext/extended_dynamic_state.rs

```diff
@@ -0,0 +1,179 @@
+#![allow(dead_code)]
+use crate::version::{DeviceV1_0, InstanceV1_0};
+use crate::vk;
+use std::ffi::CStr;
+use std::mem;
+
+#[derive(Clone)]
+pub struct ExtendedDynamicState {
+    handle: vk::Instance,
+    extended_dynamic_state_fn: vk::ExtExtendedDynamicStateFn,
+}
+
+impl ExtendedDynamicState {
+    pub fn new<I: InstanceV1_0, D: DeviceV1_0>(instance: &I, device: &D) -> Self {
+        let extended_dynamic_state_fn = vk::ExtExtendedDynamicStateFn::load(|name| unsafe {
+            mem::transmute(instance.get_device_proc_addr(device.handle(), name.as_ptr()))
+        });
+        ExtendedDynamicState {
+            handle: instance.handle(),
+            extended_dynamic_state_fn,
+        }
+    }
+
+    pub fn name() -> &'static CStr {
+        vk::ExtExtendedDynamicStateFn::name()
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetCullModeEXT.html>"]
+    pub unsafe fn cmd_set_cull_mode_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        cull_mode: vk::CullModeFlags,
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_cull_mode_ext(command_buffer, cull_mode);
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetFrontFaceEXT.html>"]
+    pub unsafe fn cmd_set_front_face_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        front_face: vk::FrontFace,
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_front_face_ext(command_buffer, front_face)
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetPrimitiveTopologyEXT.html>"]
+    pub unsafe fn cmd_set_primitive_topology_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        primitive_topology: vk::PrimitiveTopology,
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_primitive_topology_ext(command_buffer, primitive_topology)
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetViewportWithCountEXT.html>"]
+    pub unsafe fn cmd_set_viewport_with_count_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        viewport_count: u32,
+        p_viewports: *const vk::Viewport,
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_viewport_with_count_ext(command_buffer, viewport_count, p_viewports)
```

Comment:
These hand-written wrappers accept safe types and convert them to their unsafe variant. This should take a slice of `viewports: &[vk::Viewport]` and pass the `.len()` and `.as_ptr()` into the unsafe function.

Reply:
Will do. I did a copy/paste of all the functions and forgot to make them safer...